src/Widgets/Popovers/AvatarPopover.vala

```diff
@@ -78,20 +80,11 @@ namespace SwitchboardPlugUserAccounts.Widgets {
             file_dialog.set_preview_widget (preview_area);
             file_dialog.update_preview.connect (() => {
                 string uri = file_dialog.get_preview_uri ();
-                // We only display local files:
-                if (uri != null && uri.has_prefix ("file://") == true) {
-                    try {
-                        Gdk.Pixbuf pixbuf = new Gdk.Pixbuf.from_file_at_scale (uri.substring (7), 150, 150, true);
-                        preview_area.set_from_pixbuf (pixbuf);
-                        preview_area.show ();
-                        file_dialog.set_preview_widget_active (true);
-                    } catch (Error e) {
-                        preview_area.hide ();
-                        file_dialog.set_preview_widget_active (false);
-                    }
+                if (uri != null && uri.has_prefix ("file://")) {
+                    preview_area.set_from_gicon_async.begin (new FileIcon (file_dialog.get_file ()), 256);
```

Comment:
Unhandled error may be thrown.  Should hide preview area if image not successfully loaded (or file is not an image file) in order to maintain current behaviour.